src/main/java/BaseballGame.java

```diff
@@ -0,0 +1,51 @@
+public class BaseballGame {
+    private final int LENGTH = 3;
+    private final int MIN_VALUE = 1;
+    private final int MAX_VALUE = 9;
+    private final NumberGenerator numberGenerator;
+    private final BaseballView baseballView;
+    private boolean isTerminated;
+    private int[] answerNumbers;
+
+    BaseballGame(NumberGenerator numberGenerator, BaseballView baseballView) {
+        this.numberGenerator = numberGenerator;
+        this.baseballView = baseballView;
+
+        this.isTerminated = false;
+    }
+
+    public void run() {
+        initialize();
+        while (!isTerminated) {
+            String userNumber = baseballView.inputUserNumber();
+            if (!baseballView.validateUserNumber(userNumber, LENGTH, MIN_VALUE, MAX_VALUE)) {
+                baseballView.printError();
+                continue;
+            }
+            int[] userNumbers = baseballView.refineUserNumber(userNumber);
+
+            BaseballScore score = BaseballScore.calculateScore(userNumbers, answerNumbers);
+            baseballView.printResult(score);
+
+            if (checkGameEnd(score)) {
+                handleGameEnd();
+            }
+        }
+    }
+
+    private void handleGameEnd() {
+        baseballView.printWinMessage(LENGTH);
+        isTerminated = baseballView.askGameEnd();
+        if (!isTerminated) {
+            initialize();
+        }
+    }
+
+    private void initialize() {
+        answerNumbers = numberGenerator.generateNumbers(LENGTH, MIN_VALUE, MAX_VALUE);
+    }
+
+    private boolean checkGameEnd(BaseballScore score) {
+        return score.getStrikeCount() == LENGTH;
```

Comment:
지금까지는 get을 통해 가져와서 절차지향적으로 하는 것이 익숙했을 텐데요.
직접 객체로부터 값을 가져오기보다, 물어봅시다!
```suggestion
        return score.isOut();
```

Reply:
그렇게 된다면 이 메서드도 불필요해지겠죠 ?